schema/deploy/swrs_functions/refresh_swrs_version_data.sql

```diff
@@ -53,30 +53,23 @@ begin
           raise notice 'Created new swrs version (version 0) for application ID: %', application_temp_row.id;
 
       -- Do nothing if a swrs report does not exist
-      elsif (report_imported_at is not null) then
+      elsif (generated_report_id is not null and generated_report_id != application_temp_row.report_id) then
```

Comment:
I am actually not quite sure that using the report_id is the best way to do that, since the `report_id` is defined by the ordering of the reports in the [materialized view, which has no `order by` clause](https://github.com/bcgov/cas-ggircs/blob/develop/deploy/swrs/transform/materialized_view/report.sql), and is thus unpredictable. I am not even sure that we should be using the `report_id` anywhere... Looking at our prod data, we have multiple applications with the same `report_id` (from different facility/organisation).

The `swrs.report` table does have a `version` column, which could be a better alternative

Reply:
Would this mean adding a version column to the application table to compare with swrs.report.version? Is that not confusing with our use of `version` for application 'revisions' across the CIIP portal? I could fall back to the original imported_at <-> updated_at comparison to decide whether a set of form results needs updating.

Reply:
`application.swrs_report_version` should prevent any confusion